digital-ocean.md

```diff
@@ -0,0 +1,168 @@
+# Deploying to DigitalOcean
+
+This guide will walk you through deploying SwiftNIO's [example HTTP server](https://github.com/apple/swift-nio/tree/master/Sources/NIOHTTP1Server) to a [Droplet](https://www.digitalocean.com/products/droplets/). To follow this guide, you will need to have a [DigitalOcean](https://www.digitalocean.com) account with billing configured.
+
+## Create Server
+
+Let's start by installing Swift on an Ubuntu server. Use the create menu to create a new Droplet.
+
+![Create Droplet](images/digital-ocean-create-droplet.png)
+
+Under distributions, select Ubuntu 18.04 LTS.
+
+![Ubuntu Distro](images/digital-ocean-distributions-ubuntu-18.png)
+
+> Note: You may select any version of Ubuntu that Swift supports. At the time of writing, Swift 5.2 supports 16.04 and 18.04. You can check which operating systems are officially supported on the [Swift Releases](https://swift.org/download/#releases) page.
+
+After selecting the distribution, choose any plan and datacenter region you prefer. Then setup an SSH key to access the server after it is created. Finally, click create Droplet and wait for the new server to spin up.
+
+Once the new server is ready, hover over the Droplet's IP address and click copy.
+
+![Droplet List](images/digital-ocean-droplet-list.png)
+
+## Initial Setup
+
+Open your terminal and connect to the server as root using SSH.
+
+```sh
+ssh root@your_server_ip
+```
+
+DigitalOcean has an in-depth guide for [initial server setup on Ubuntu 18.04](https://www.digitalocean.com/community/tutorials/initial-server-setup-with-ubuntu-18-04). This guide will quickly cover the basics.
+
+### Configure Firewall
+
+Allow OpenSSH through the firewall and enable it.
+
+```sh
+ufw allow OpenSSH
+ufw enable
+```
+
+### Add User
+
+Create a new user besides `root`. This guide calls the new user `swift`.
+
+```sh
+adduser swift
+```
+
+Allow the newly created user to use `sudo`.
```

Comment:
I assume this user will be the one that runs the service? if so, I am not sure this is ideal from a security pov. is it possible to do all the installations as root but run the program as swift?

Reply:
Good point. I'll try that out.

Reply:
If the `swift` user is not part of the sudo group, it will not be able to run applications on port 80. The way around this is with some reverse proxy like nginx (which I plan to do a guide on soon), but since that guide doesn't exist yet I'm not sure what to do here.

Reply:
the typical thing to do is not bind to port 80 but rather to something higher like 8080. this is fine for the purposes of this guide I would think, since when you put a LB or ngninx in front you would do the 80 -> 8080 etc

Reply:
@tanner0101 wdyt?

Reply:
Sorry I missed that last comment. Make sense. I'll swap it to use `8080` everywhere instead. Then they can follow the Nginx guide if they want port `80`.